plugins/catalog-backend/src/descriptors/types.ts

```diff
@@ -14,22 +14,24 @@
  * limitations under the License.
  */
 
-import fs from 'fs-extra';
-import { ComponentDescriptor, parseDescriptor } from '../descriptors';
+import { ComponentDescriptorV1 } from './model/ComponentDescriptorV1';
 
-export async function readFileLocation(
-  target: string,
-): Promise<ComponentDescriptor[]> {
-  let rawYaml;
-  try {
-    rawYaml = await fs.readFile(target, 'utf8');
-  } catch (e) {
-    throw new Error(`Unable to read "${target}", ${e}`);
-  }
+export type ComponentDescriptor = ComponentDescriptorV1;
 
-  try {
-    return parseDescriptor(rawYaml);
-  } catch (e) {
-    throw new Error(`Malformed descriptor at "${target}", ${e}`);
-  }
-}
+export type DescriptorEnvelope = {
+  apiVersion: string;
+  kind: string;
+  metadata?: object;
+  spec?: object;
+};
+
+export type DescriptorParser = {
+  apiVersion: string;
+  kind: string;
+  parse(envelope: DescriptorEnvelope): Promise<ParserOutput>;
+};
+
+export type ParserOutput = {
+  errors: Error[];
+  components: ComponentDescriptor[];
```

Comment:
Think we really need support for multiple components here? Thinking something like this instead:

```ts
type ParseResult = {
  error: Error
} | {
  component: ComponentDescriptor
}
```

Reply:
Composite readers may output several things AND errors at the same time.

Reply:
What input would produce multiple components?

Reply:
This is also the output type of LocationReaders.

Reply:
So, yaml files.

Reply:
Ah didn't get that you wanted to reuse the type. Was about to wonder about how you'd know which component had an error. I'd say have a separate type for the `DescriptorParser` result then tbh. Or just make it return `ComponentDescriptor` and reject the promise instead.